services/datamanager/builtin/capture_test.go

```diff
@@ -268,8 +268,7 @@ func TestSwitchResource(t *testing.T) {
 		// Assert that we see the expected data captured by the initial arm1 resource.
 		test.That(
 			t,
-			d.GetStruct().GetFields()["Number"].GetStructValue().GetFields()["Dual"].GetStructValue().GetFields()["Jmag"].GetNumberValue(),
-			test.ShouldEqual,
+			d.GetStruct().GetFields()["pose"].GetStructValue().GetFields()["o_z"].GetNumberValue(), test.ShouldEqual,
```

Comment:
Surprised we don't need to change more tests - or do we not have tests on the output of each collector? I wonder if we should add that as part of this ticket (I think yes). 

Idea for a simple and concise way to test the collectors: we could call a particular method using gRPC and also via a collector and check that the response formats are the same.

Reply:
@dmhilly @AaronCasas I talked to Alexa about the testing for collectors in general and this is the outcome:
Adding tests is definitely something we should do but is lower priority than project work. I'll make a ticket for testing and work on that separately maybe down the line but prioritizing whatever project work is there(which I know is how we got here LOL).

Reply:
https://viam.atlassian.net/browse/DATA-2012

Reply:
We should make sure to test every changed collector, whether manually or via an automated test suite — initially assumed the former would be faster, but Devin's suggestion seems like the latter would likely be faster. Before merging these changes, it would actually be worth adding these tests to ensure we're only syncing data in the format that we expect.

Reply:
As discussed offline, whatever can be done within a day of focused time before merging these changes, which seems possible with the suggested testing methodology!